src/QsCompiler/Transformations/CallGraph.cs

```diff
@@ -1,47 +1,95 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
 using Microsoft.Quantum.QsCompiler.DataTypes;
-using Microsoft.Quantum.QsCompiler.SyntaxTokens;
 using Microsoft.Quantum.QsCompiler.SyntaxTree;
-using Microsoft.Quantum.QsCompiler.Transformations.Core;
+using Microsoft.Quantum.QsCompiler.Transformations.CallGraphWalker;
 
 #nullable enable
 
-// ToDo: Review access modifiers
-
 namespace Microsoft.Quantum.QsCompiler.DependencyAnalysis
 {
-    using ResolvedTypeKind = QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>;
-    // Type Parameters are frequently referenced by the callable of the type parameter followed by the name of the specific type parameter.
-    using TypeParameterName = Tuple<QsQualifiedName, NonNullable<string>>;
-    using TypeParameterResolutions = ImmutableDictionary</*TypeParameterName*/ Tuple<QsQualifiedName, NonNullable<string>>, ResolvedType>;
+    using TypeParameterResolutions = ImmutableDictionary<Tuple<QsQualifiedName, NonNullable<string>>, ResolvedType>;
+
+    /// <summary>
+    /// Base class for call graph edge types.
+    /// </summary>
+    public abstract class CallGraphEdgeBase : IEquatable<CallGraphEdgeBase>
```

Comment:
There's a lot happening in this file that makes it a bit hard for me to follow: several top-level classes from two different call graphs plus the base call graph, and several nested classes. Could this be split out into more than one file? The base call graph, simple call graph, and concrete call graph seem like they could go in 3 separate files.

Reply:
The three call graphs are now in there own separate files.